xapian-bindings/java/java.i

```diff
@@ -174,21 +174,63 @@ namespace Xapian {
     bool hasNext() const { return Xapian::iterator_valid(*self); }
 }
 
+%ignore MSetIterator::next();
+%javamethodmodifiers MSetIterator::next_helper() "private"
+
 %extend MSetIterator {
-    Xapian::docid next () {
-	Xapian::docid tmp;
+    void next_helper () {
 	if (Xapian::iterator_valid(*self)) {
```

Comment:
I'm not sure why this is a separate function; it's only used once, so why not just inline it in the `next()` implementation further down?